msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-get-started.md

```diff
@@ -0,0 +1,608 @@
+  
+* `config.json`: Provide the right configuration to ensure bot responses are aligned with your requirement. Configure `max_tokens`, `temperature`, and other properties to pass into open AI or Azure AI. Contains the prompt model settings.
+
+   ```json
+   {
+    "schema": 1,
+    "description": "Chat with Santa Clause",
+    "type": "completion",
+    "completion": {
+      "max_tokens": 150,
+      "temperature": 0.9,
+      "top_p": 0.0,
+      "presence_penalty": 0.6,
+      "frequency_penalty": 0.0,
+      "stop_sequences": [
+        "Human:",
+        "AI:"
+      ]
+    }
+   }
+   ```
+
+### Query parameters
+
+The following table includes the query parameters:
+
+|**Value**  |**Description**  |
+|---------|---------|
+|`max_tokens`     | The maximum number of tokens to generate in the completion. The token count of your prompt plus max_tokens cannot exceed the model's context length.        |
+|`temperature`    | What sampling temperature to use, between 0 and 2. Higher values like 0.8 will make the output more random, while lower values like 0.2 will make it more focused and deterministic.        |
+|`top_p`    |An alternative to sampling with temperature, called nucleus sampling, where the model considers the results of the tokens with top_p probability mass. So 0.1 means only the tokens comprising the top 10% probability mass are considered.         |
+|`presence_penalty`     |  Number between -2.0 and 2.0. Positive values penalize new tokens based on whether they appear in the text so far, increasing the model's likelihood to talk about new topics.       |
+|`frequency_penalty`     |Number between -2.0 and 2.0. Positive values penalize new tokens based on their existing frequency in the text so far, decreasing the model's likelihood to repeat the same line verbatim.         |
+|`stop_sequences`     |  Up to 4 sequences where the API will stop generating further tokens. The returned text will not contain the stop sequence. |
+
+### Prompt actions
+
+Plans let the model perform actions or say things to the user. You can create a schema of the plan and add a list of actions that you support. It can perform an action and pass arguments. GPT can  figure out what actions it wants to use and then extract all the entities and pass those in as arguments to the action call.
+
+```text
+    The following is a conversation with an AI assistant. 
+    The AI is Santa Clause and the Human is a child meeting Santa for the first time. 
+    The AI should always reply the way Santa would. 
+    The AI should always greet the human the way Santa would, ask them their name, and then what they would like for Christmas.
+    
+    {{$history}}
+    Human: {{$input}}
+    AI:
+  ```
+
+### Prompt Template
+
+You can add functions to call a callback and return any kind of data you want.
+
+* {{function}}:  Calls a registered function and inserts its value.​
+
+* {{$input}}:  Inserts the message text.​
```

Comment:
It's worth mentioning that this is basically `state.temp.input`

and same for history, it's `state.temp.history`

Reply:
Updated as per suggestions provided above.